### PR TITLE
Feature/essql search strategy exp

### PR DESCRIPTION
**Pull request by poffdeluxe**

### CODE REVIEW

File: x-pack/plugins/canvas/public/services/expressions.ts [hunk 1]
```diff
@@ -32,9 +37,15 @@ export const expressionsServiceFactory: CanvasServiceFactory<ExpressionsService>
         // function that matches its definition, but which simply
         // calls the server-side function endpoint.
         Object.keys(serverFunctionList).forEach((functionName) => {
-          if (expressions.getFunction(functionName)) {
+          // Allow function to be overwritten if we want to use
+          // the server-hosted essql, esdocs, and escount functions
+          if (
+            !(!hasDataSearch && dataSearchFns.includes(functionName)) &&
+            expressions.getFunction(functionName)
```
Comment (poffdeluxe) [inline x-pack/plugins/canvas/public/services/expressions.ts:44]: This is where the magic happens. On the server, we still load up esql, escount, and esdocs. We load up those three functions (using the data search stuff) on client as well. If we have the data search experiment enabled, we will skip the function and let the client-side version run. If the experiment is disabled, we overwrite the new client version of the function with the function that calls out to the server

Comment (poffdeluxe) [reply x-pack/plugins/canvas/public/services/expressions.ts:44]: as a side note, i'm gonna break this if-statement into an if-else for readability. It's too hard to parse at the moment

Comment (clintandrewhall) [reply x-pack/plugins/canvas/public/services/expressions.ts:44]: Rock ON, man.  Epic... I like this.

File: x-pack/plugins/canvas/public/services/stubs/search.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// import { SearchService } from '../search';
```
Comment (clintandrewhall) [inline x-pack/plugins/canvas/public/services/stubs/search.ts:8]: nit

File: src/plugins/presentation_util/public/index.ts [hunk 3]
```diff
@@ -11,6 +11,8 @@ import { pluginServices } from './services';
 
 export { PresentationUtilPluginSetup, PresentationUtilPluginStart } from './types';
 
+export { experimentIDs, Experiment, ExperimentID } from '../common';
```
Comment (clintandrewhall) [inline src/plugins/presentation_util/public/index.ts:14]: I resolved this in the parent PR, thankfully.  Let me know if you have other instances where things didn't line up.

File: x-pack/plugins/canvas/public/components/workpad_manager/workpad_manager.js [hunk 4]
```diff
@@ -55,6 +57,13 @@ export const WorkpadManager = ({ onClose }) => {
               <h1>{strings.getModalTitle()}</h1>
             </EuiModalHeaderTitle>
           </EuiFlexItem>
+          <EuiFlexItem>
+            <div>
+              <Suspense fallback="Loading...">
```
Comment (clintandrewhall) [inline x-pack/plugins/canvas/public/components/workpad_manager/workpad_manager.js:62]: Fixed this, too, in the parent PR with an HOC: `withSuspense`

**Review by clintandrewhall — COMMENTED** [review]
This is great, Poff.  Let's get the experiment service reviewed so we can get this in.  Feel free to merge the latest changes in the meantime?
